pkg/utils/tls/tls_test.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2025 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tls
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"os"
+	"testing"
+	"time"
+)
+
+func generateTestCert(t *testing.T, validFor time.Duration) (certFile, keyFile string) {
+	t.Helper()
+
+	certFile = mustTempFile(t)
+	keyFile = mustTempFile(t)
```

Comment:
What about using https://pkg.go.dev/testing#T.TempDir instead so you don't need to worry about the cleanup?